src/index.tsx

```diff
@@ -1,26 +1,20 @@
-import * as React from "react";
-import * as ReactDOM from "react-dom";
-import { AppContainer } from "react-hot-loader";
-import App from "./App";
+import { AppContainer } from 'react-hot-loader';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import {App} from './App';
 
-const rootEl = document.getElementById("root");
-ReactDOM.render(
+const rootEl = document.getElementById('root');
+
+function render() {
+  ReactDOM.render(
     <AppContainer>
-        <App />
+      <App />
     </AppContainer>,
     rootEl
-);
+  );
+}
+
+render();
 
-// Hot Module Replacement API
-if (module.hot) {
-  module.hot.accept("./App", () => {
-    const NextApp = require("./App").default;
-    ReactDOM.render(
-      <AppContainer>
-        <NextApp/>
-      </AppContainer>
-      ,
-      rootEl
-    );
-  });
-}
\ No newline at end of file
+// module.hot?.accept('./App', render);
```

Comment:
actually, this is not a "commented out code", but the explanation for the following code. It means that "if `module.hot` is not `undefined` then I want to invoke method `accept` of it". It's definitely not "commented out code" because it's still not syntactically valid in typescript, although lots people have requested https://github.com/Microsoft/TypeScript/issues/16